### PR TITLE
Issue 299 mem leak

### PR DESCRIPTION
**Pull request by danielwertheim**
A first fix for #299 ensuring a token registration is cleaned up

Without this fix, MB consumption when monitoring in Visual Studio keeps going up. Will look further into that request construct but wanted to provide this fix first.

### CODE REVIEW

**Review by watfordgnf — APPROVED** [review]
Ack, I forgot to submit this fix. Looks good!

File: src/Samples/WinFormsSample/Form1.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            _pubConnection?.Dispose();
+            _pubConnection = null;
+
+            _cts.Cancel();
+
+            _subConnection?.Dispose();
+            _subConnection = null;
+        }
+
+        private async void btnRun_Click(object sender, EventArgs e)
```
Comment (ColinSullivan1) [inline src/Samples/WinFormsSample/Form1.cs]: Think it'd be a good idea to disable the button while running, or do you want to allow multiple parallel runs?

File: src/Samples/WinFormsSample/Form1.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            {
+                var payload = Encoding.UTF8.GetBytes(Guid.NewGuid().ToString("N"));
+
+                var numOfMessages = numMessages.Value;
+                var configAwaitFalse = chkConfigureAwaitFalse.Checked;
+
+                for (var i = 0; i < numOfMessages; i++)
+                    await _pubConnection.RequestAsync(Subject, payload, _cts.Token).ConfigureAwait(!configAwaitFalse);
+            }));
+
```
Comment (ColinSullivan1) [inline src/Samples/WinFormsSample/Form1.cs:86]: Small request - could you initialize one of the list items?  e.g.
`lstScenarios.SelectedItem = lstScenarios.Items[0];`

**Comment by ColinSullivan1** [comment]
Edited:  Other than a few minor things and the windows form issue, LGTM.  Forms brings back some old memories.  :)

**Review by ColinSullivan1 — APPROVED** [review]

**Review by ColinSullivan1 — CHANGES_REQUESTED** [review]
In other testing, it looks like the windows forms project prevents builds on linux and mac.  Could you make that project only build on the windows platform?

**Comment by danielwertheim** [comment]
@ColinSullivan1 I've now fixed so that all projects can be built in a Linux Docker container and I hope that is good enough verification. Had to exclude explicit support for `.NET4.5` for **non-windows**. Is that OK?

It would be simpler if we stuck to .NET Standard and .NET Core app only but this works for now. To not build the Forms app I had to include a `LinuxRelease` and `LinuxDebug` build config do define that WinForms project should not be built then.

We could look at `LinuxDebug` and `LinuxRelease` instead of:

```
$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Windows)))
```

That way you control what to build in what env. Perhaps a cleaner approach? That would look like this: https://github.com/nats-io/nats.net/commit/4c0fd9a44319fc52620b846c554f9fc45b53e3b8

**Comment by ColinSullivan1** [comment]
Thanks @danielwertheim.  

>  Had to exclude explicit support for .NET4.5 for non-windows. Is that OK?

Absolutely - the alternative is mono, which imo should be a custom build on a fork at this point.

I would like to keep .NET 4.5 for the time being, there are some users that still require that framework.

> We could look at LinuxDebug and LinuxRelease...

That sounds cleaner to me - might want to add a line in the readme about selecting Linux to building on OSx - we have some users that build .NET there.

**Review by ColinSullivan1 — APPROVED** [review]
LGTM - Thanks!

**Comment by danielwertheim** [comment]
@ColinSullivan1  I'll add the config solution to this PR and complete/merge it when done.

**Comment by ColinSullivan1** [comment]
Thanks!

**Pull request merged**